core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -682,13 +682,35 @@ public boolean isSnapshotting() {
   public void notifyLeaderChanged(RaftGroupMemberId groupMemberId, RaftPeerId raftPeerId) {
     if (mRaftGroupId == groupMemberId.getGroupId()) {
       mIsLeader = groupMemberId.getPeerId() == raftPeerId;
+      if (!mIsLeader) {
+        // If this state machine wins an election right after starting or right after losing
+        // another election, it will not trigger notifyExtendedNoLeader. This is therefore the
+        // last point of contact with another leader.
+        mJournalSystem.setLastLeaderTime(System.currentTimeMillis());
```

Comment:
what if the state-machine was not the leader to being with? are we then setting last-leader-time of a particular state-machine ahead of when it was a leader the last time?

Reply:
See response to [this comment](https://github.com/Alluxio/alluxio/pull/14572/files#r768952525).